libs/entitlement/qa/entitlement-test-plugin/src/main/java/org/elasticsearch/entitlement/qa/test/URLConnectionNetworkActions.java

```diff
@@ -79,6 +79,17 @@ private static void withJdkHttpConnection(CheckedConsumer<HttpURLConnection, Exc
         }
     }
 
+    private static void withJdkMailToConnection(CheckedConsumer<URLConnection, Exception> connectionConsumer) throws Exception {
+        var conn = EntitledActions.createMailToURLConnection();
+        // Be sure we got the connection implementation we want
+        assert conn.getClass().getSimpleName().equals("MailToURLConnection");
```

Comment:
👍  Not much of an alternative here. At least it's good to know this test would fail if run on a JDK that did not use this implementation.

Reply:
Or if in a future JDK this changes name or implementation (that was exactly my intention, so the test fails and we look at it)